test/common_utils.py

```diff
@@ -16,3 +17,34 @@ def create_temp_assets_dir():
     copytree(os.path.join(TEST_DIR_PATH, "assets"),
              os.path.join(tmp_dir.name, "assets"))
     return tmp_dir.name, tmp_dir
+
+
+class RandomTensorGenerator:
+    """ Generates random tensors given a seed and size
+    https://en.wikipedia.org/wiki/Linear_congruential_generator
+    X_{n + 1} = (a * X_n + c) % m
+    Using Borland C/C++ values
+    """
+    a = 22695477
+    c = 1
+    m = 2 ** 32
+
+    def __init__(self, seed, size):
+        # seed is an int and size is a tuple of ints
+        num_elements = 1
+        for s in size:
+            num_elements *= s
+
+        arr = [seed]
```

Comment:
seed is an int64_t if its a python integer, so it can't be the first number if you always mod by m for the other values.